src/App.tsx

```diff
@@ -15,71 +17,75 @@ export const goods = [
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  render() {
+    const { selectedGood } = this.state;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        {!selectedGood && (
+          <h1 className="title">
+            No goods selected
+          </h1>
+        )}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+        {selectedGood && (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={() => (this.setState({ selectedGood: '' }))}
+            />
+          </h1>
+        )}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              return (
+                <tr
+                  data-cy="Good"
+                  className={classNames(
+                    { 'has-background-success-light': selectedGood === good },
+                  )}
+                >
+                  <td>
+                    <button
+                      data-cy={selectedGood === good
+                        ? 'RemoveButton'
+                        : 'AddButton'}
+                      type="button"
+                      className={
+                        classNames(
+                          'button',
+                          { 'is-info': selectedGood === good },
+                        )
+                      }
+                      onClick={() => (this.setState({ selectedGood: good }))}
```

Comment:
Add an 'onClick' event handler for the 'RemoveButton' to remove the selection when clicked.